File: src/main/java/com/dansoftware/boomega/i18n/InternalLanguagePack.java
/*
 * Boomega - A modern book explorer & catalog application
 * Copyright (C) 2020-2022  Daniel Gyoerffy
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.dansoftware.boomega.i18n;

import com.dansoftware.boomega.util.Person;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Locale;
import java.util.ResourceBundle;

abstract class InternalLanguagePack extends LanguagePack {

    private static final String VALUES = "com.dansoftware.boomega.i18n.Values";

    protected InternalLanguagePack(@NotNull Locale locale) {
        super(locale);
    }

    @Override
    public @Nullable Person getTranslator() {
        return null;
    }

    @Override
    public @NotNull ResourceBundle getValues() {
        return super.getBundle(VALUES);
    }
}

File: src/main/java/com/dansoftware/boomega/i18n/InternalLanguagePack.kt
package com.dansoftware.boomega.i18n

import com.dansoftware.boomega.util.Person
import java.util.*

/**
 * An [InternalLanguagePack] is an abstract [LanguagePack] implementation
 * that represents a Language pack that is nested into the application by default.
 *
 * @author Daniel Gyorffy
 */
private abstract class InternalLanguagePack(locale: Locale) : LanguagePack(locale) {

    companion object {
        private const val VALUES = "com.dansoftware.boomega.i18n.Values"
    }

    override fun getTranslator(): Person? = null

    override fun getValues(): ResourceBundle = getBundle(VALUES)
}